lib/tagger.rb

```diff
@@ -0,0 +1,22 @@
+class Tagger
+  def self.add_tags(content_id, taxon_ids, do_tag = true)
+    link_content = Services.publishing_api.get_links(content_id)
+    version = link_content['version']
+    existing_taxon_ids = link_content.dig('links', 'taxons') || []
```

Comment:
You could extract these lines out to a method, e.g. `get_existing_tagged_taxons` or something to make this a bit more readable.

Reply:
done - refactored this method